### DISCUSSION TITLE
Sidebar: Add permanent visible top and bottom container slot

### DISCUSSION THREAD

**Discussion by felldo** [primefaces/community#577] [PrimeVue] [answered]
I have a sidebar which can exceed the height of the page with its content (see image). The sidebar automatically adds a scrollbar to be able to see the other content through scrolling. The header stays at the position as it is and does not move.
Currently, there is no easy/nice way to set a top/bottom element which also stays like the header and does not move/scrolls with the content.
I think this might be a common case where someone wants a header / footer in a sidebar where something is always displayed (might be a settings button which is common in the footer of a sidebar), even if there are many elements, and a scrollbar is added to scroll through this content. The current header is not suited for this purpose, because it aligns badly with the close icon.
Therefore, I suggest adding a top/bottom container slot which behaves like the current header and is always visible but completely customizable by ourselves.

Images for reference:
At the top of the scrollable content. The "Speichern" -> "Save" Button is not visible, because the content is too long.
![image](https://github.com/primefaces/community/assets/31566807/f6634316-1b95-4f5c-bf33-4d729abd85a3)
Scrolled to the bottom. The description at the very top and the Buttons are no longer visible.
![image](https://github.com/primefaces/community/assets/31566807/a38ad6c6-4534-4d12-93a2-eef162d86557)

**Comment by tugcekucukoglu** [comment, marked as answer]
If you think that the header slot does not behave as you want, we can say container slot as an alternative. An example of this would be [headless mode](https://primevue.org/sidebar/#headless) in the live demo.

**Reply by felldo** [reply]
I am aware of the headless mode, but this isn't really an easy/nice way to do this. I don't think it would hurt or clutter this library if 2 new slots would be defined as they might be pretty common use cases, especially the footer slot, where often some sort of user profile content is which often leads to settings or something similar. Just take Discord as an example. Your user profile is always visible at the bottom regardless of where you currently are (in a server or DMs). Therefore I think adding these 2 slots would only benefit PrimeVue with being an even more customizable library :)
